contracts/land/LANDRegistry.sol

```diff
@@ -200,18 +223,76 @@ contract LANDRegistry is Storage,
     emit UpdateOperator(assetId, operator);
   }
 
+  //
+  // Estate generation
+  //
+
+  event EstateRegistrySet(address indexed registry);
+
+  function setEstateRegistry(address registry) onlyProxyOwner external {
+    estateRegistry = IEstateRegistry(registry);
+    emit EstateRegistrySet(registry);
+  }
+
+  function createEstate(
+    int[] x,
+    int[] y,
+    address beneficiary
+  )
+    external
+    returns (uint256)
+  {
+    require(x.length == y.length);
```

Comment:
What do you think of checking also if the length is greater than 0?

`require(x.length > 0 && x.length == y.length);`